taskweaver/llm/openai.py

```diff
@@ -98,6 +98,11 @@ def _configure(self) -> None:
         self.presence_penalty = self._get_float("presence_penalty", 0)
         self.seed = self._get_int("seed", 123456)
 
+        self.require_alternative_roles = self._get_bool("require_alternative_roles", False)
+        self.support_system_role = self._get_bool("support_system_role", True)
+        self.support_constrained_generation = self._get_bool("support_constrained_generation", False)
+        self.json_schema_enforcer = self._get_str("json_schema_enforcer", None, required=False)
+
 
```

Comment:
Considering these extra things are not supported by official OpenAI endpoints, do we need to put them in this configuration class? Or is it possible to have a separate LLM configuration for supporting these flags?

Reply:
A number of model serving platforms support this OpenAI compatible API. If we move them into a dedicated configuration class, we would need them to configure something like `openai_compatible`?